### PR TITLE
methods 'pull_tbl.dm()' and 'pull_tbl.zoomed_dm()'

### LINKED ISSUE #206: How to extract the currently zoomed table?
Use case: zoom, summarize, join, then check result.

Should work for both in-memory and database dm-s.

### PR DESCRIPTION
**Pull request by TSchiefer**
- methods `pull_tbl.dm()` and `pull_tbl.zoomed_dm()`
- documentation in addition to `?pull_tbl` also in zoom-vignette

closes #206

### CODE REVIEW

File: R/dm.R [hunk 1]
```diff
@@ -687,3 +687,40 @@ empty_dm <- function() {
     )
   )
 }
+
+#' Retrieve a table from a `dm` or a `zoomed_dm`
+#'
+#' This function has methods for both `dm` classes:
+#' 1. with `pull_tbl.dm()` you can chose which table of the `dm` you want to retrieve
+#' 1. with `pull_tbl.zoomed_dm()` you will retrieve the zoomed table in the current state
```
Comment (krlmlr) [inline R/dm.R]:
```suggestion
#' 1. with `pull_tbl.zoomed_dm()` you will retrieve the zoomed table in the current state.
```

File: R/dm.R [hunk 2]
```diff
@@ -687,3 +687,40 @@ empty_dm <- function() {
...
+#'
+#' @inheritParams dm_add_pk
+#' @param table One unquoted table name for `pull_tbl.dm()`, ignored for `pull_tbl.zoomed_dm()`
```
Comment (krlmlr) [inline R/dm.R]:
```suggestion
#' @param table One unquoted table name for `pull_tbl.dm()`, ignored for `pull_tbl.zoomed_dm()`.
```

File: R/dm.R [hunk 3]
```diff
@@ -687,3 +687,40 @@ empty_dm <- function() {
...
+#'
+#' @return The requested table
+#'
+#' @examples
+#' # both examples lead to the same tibble:
+#' pull_tbl(dm_nycflights13(), airlines)
```
Comment (krlmlr) [inline R/dm.R]: Can you please start the pipe with `dm_nycflights13()`?

File: R/dm.R [hunk 4]
```diff
@@ -687,3 +687,40 @@ empty_dm <- function() {
...
+pull_tbl.dm <- function(dm, table) {
+  # FIXME: shall we issue a special error in case someone tries sth. like: `pull_tbl(dm_for_filter, c(t4, t3))`?
+  table_name <- as_string(enexpr(table))
+  if (table_name == "") abort_no_table_provided()
+  tbl(dm, table_name)
+}
+
+#' @export
+pull_tbl.zoomed_dm <- function(dm, table) {
+  if (!is_missing(table)) warning("Ignoring argument `table` in `pull_tbl.zoomed_dm`, returning zoomed table.")
```
Comment (krlmlr) [inline R/dm.R]: Can we check that `table` is one of the zoomed tables (currently "the" zoomed table), and err otherwise?

File: R/error-helpers.R [hunk 5]
```diff
@@ -567,3 +567,10 @@ abort_dm_invalid <- function(why) {
 error_dm_invalid <- function(why) {
   paste0("This `dm` is invalid, reason: ", why)
 }
+
+
+# Argument `table` missing in `pull_tbl.dm()` -----------------------------
+
+abort_no_table_provided <- function() {
```
Comment (krlmlr) [inline R/error-helpers.R]: Can you please update the file that shows all error messages?

File: R/dm.R (hunk 4 above)
Comment (TSchiefer) [reply R/dm.R]: also if `table` is not provided?

Comment (krlmlr) [reply R/dm.R]: Only if provided, by default we use "the" zoomed table.

Comment (TSchiefer) [reply R/dm.R]: Implemented it now (with regard to possible future developments) in a way, that if more than one zoomed table is available and argument `table` is empty, an error is issued.

**Comment by krlmlr** [comment]
Thanks!

**Pull request merged**